packages/website/pages/files.js

```diff
@@ -96,6 +101,118 @@ export default function Files({ user }) {
 
   const hasZeroNfts = nfts.length === 0 && befores.length === 1
 
+  /**
+   * @param {any} nft
+   */
+  const TableItem = ({ nft }) => {
+    // to do, add actual types
+    const deals = nft.deals
+      .filter((/** @type {any} */ d) => d.status !== 'queued')
+      .map(
+        (
+          /** @type {any} */ deal,
+          /** @type {number} */ i,
+          /** @type {any[]} */ deals
+        ) => {
+          const url = `https://filfox.info/en/deal/${deal.chainDealId}`
+          return (
+            <span key={deal.chainDealId} title={deal.status}>
+              <a
+                className="underline"
+                href={url}
+                target="_blank"
+                rel="noreferrer"
+              >
+                {deal.miner}
+              </a>
+              {i === deals.length - 1 ? '' : ', '}
+            </span>
+          )
+        }
+      )
+
+    const queuedDeals = nft.deals.filter(
+      (/** @type {any} */ d) => d.status === 'queued'
+    )
+    if (queuedDeals.length) {
+      const message = `The content from this upload has been aggregated for storage on Filecoin and is queued for deals with ${
+        queuedDeals.length
+      } storage provider${
+        queuedDeals.length > 1 ? 's' : ''
+      }. Filecoin deals will be active within 48 hours of upload.`
+      deals.push(
+        <span key={nft.cid + '-pending'}>
+          {`${deals.length ? ', ' : ''}${queuedDeals.length} pending`}
+          <Tooltip
+            placement="top"
+            overlay={<span>{message}</span>}
+            overlayClassName="table-tooltip"
+          >
+            <VscQuestion size={16} className="ml2" />
+          </Tooltip>
+        </span>
+      )
+    }
+
+    if (!nft.deals.length) {
```

Comment:
Having some if statements mutate an array seems iffy, this is subjective, but it feels like we could probably
move some of this visual complexity into the returned dom and avoid the need to mutate the deals array.

Reply:
This is a refactor request, can be answered another time.